src/pldm_fw.cpp

```diff
@@ -7,33 +7,135 @@
 #include <libpldm/platform.h>
 #include <libpldm/pldm.h>
 #include <libpldm/state_set.h>
+#include <poll.h>
 
 #include <iostream>
 #include <sdbusplus/asio/connection.hpp>
 #include <sdbusplus/asio/object_server.hpp>
 
 namespace panel
 {
-types::Byte PldmFramework::getInstanceID()
+pldm_instance_id_t PldmFramework::getInstanceID()
 {
-    types::Byte instanceId = 0;
-    auto bus = sdbusplus::bus::new_default();
+    // Fetch Instance ID DB
+    int rc = pldm_instance_db_init_default(&pldmInstanceIdDb);
 
-    try
+    if (rc < 0)
+    {
+        throw std::runtime_error(
+            "Call to pldm_instance_db_init_default failed with return code " +
+            std::to_string(rc));
+    }
+
+    // Get Instance ID
+    pldm_instance_id_t instanceID = 0;
+
+    rc = pldm_instance_id_alloc(pldmInstanceIdDb, tid, &instanceID);
+    if (rc == -EAGAIN)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Not sure, how much change can happen under 100 milli seconds. Should the wait be for some considerable time? Else what is the point re-trying?

Reply:
ideally instance id allocation should not fail.. But for some reason if libpldm fails to get instance id for it's users - it returns EAGAIN asking us to try querying again. I don't find any rule in libpldm asking us to retry after a delay. This delay is for our safer side. 100ms is working fine till now. I would suggest if we hit any issues then we can modify this delay.

Reply:
I have a question on this based on your last sentence , since you said that this working fine , wanted to know how did you test this condition to be working fine ?